receiver/awsxrayreceiver/receiver.go

```diff
@@ -0,0 +1,206 @@
+// https://github.com/aws/aws-xray-daemon/blob/master/cmd/tracing/daemon.go#L257
+func (x *xrayReceiver) read(buf *[]byte) (int, error) {
+	bufVal := *buf
+	rlen, err := x.udpSock.Read(bufVal)
+	if err == nil {
+		return rlen, nil
+	}
+	switch err := err.(type) {
+	case net.Error:
+		if !err.Temporary() {
+			return -1, fmt.Errorf("read from UDP socket: %w", &errIrrecoverable{err})
+		}
+	default:
+		return 0, fmt.Errorf("read from UDP socket: %w", &errRecoverable{err})
+	}
+	return 0, fmt.Errorf("read from UDP socket: %w", &errRecoverable{err})
+}
+
+// this function references the implementation in:
+// https://github.com/aws/aws-xray-daemon/blob/master/cmd/tracing/daemon.go#L275
+// However, it intentionally avoids using a buffer pool and just stick to
+// a local buffer for simplicity and improve concurrency (because in the orignal
+// implementation, the buffer pool is shared between 2 poll() cals executed by
+// 2 goroutines, which requires locking/unlocking a sync.Mutex whenever
+// a buffer is Get()/Return()). Also, the buffer returns by Get()
+// (i.e. https://github.com/aws/aws-xray-daemon/blob/d2331c8c4538e55c237b05306a3cf2e919a41ba9/pkg/bufferpool/bufferpool.go#L28)
+// is the same size as the fallBackBuffer
+// (https://github.com/aws/aws-xray-daemon/blob/master/cmd/tracing/daemon.go#L277)
+func (x *xrayReceiver) poll() {
+	defer x.wg.Done()
+	separator := []byte(util.ProtocolSeparator)
+	buffer := make([]byte, pollerBufferSizeKB)
+	splitBuf := make([][]byte, 2)
+
+	var (
+		errRecv   *errRecoverable
+		errIrrecv *errIrrecoverable
+	)
+
+	for {
+		// TODO:
+		// call https://pkg.go.dev/go.opentelemetry.io/collector@v0.4.1-0.20200622191610-a8db6271f90a/obsreport?tab=doc#StartTraceDataReceiveOp
+		// once here and
+		// https://pkg.go.dev/go.opentelemetry.io/collector@v0.4.1-0.20200622191610-a8db6271f90a/obsreport?tab=doc#EndTraceDataReceiveOp
+		// at corresponding places in the for loop below.
+		bufPointer := &buffer
+		rlen, err := x.read(bufPointer)
+		if err != nil && errors.As(err, &errIrrecv) {
+			x.logger.Error("irrecoverable socket read error. Exiting poller", zap.Error(err))
+			return
+		} else if errors.As(err, &errRecv) {
+			x.logger.Error("recoverable socket read error", zap.Error(err))
+			continue
+		}
+
+		bufMessage := buffer[0:rlen]
+
+		slices := util.SplitHeaderBody(x.logger, &bufMessage, &separator, &splitBuf)
+		if len(slices[1]) == 0 {
+			x.logger.Warn("Missing header or segment", zap.ByteString("value", slices[0]))
+			// TODO: emit metric here to indicate segment rejected
+			continue
+		}
+
+		header := slices[0]
+		// payload := slices[1]
+		headerInfo := tracesegment.Header{}
+		json.Unmarshal(header, &headerInfo)
+
+		switch headerInfo.IsValid() {
+		case true:
+		default:
+			x.logger.Warn("Invalid header", zap.ByteString("header", header))
+			// TODO: emit metric here to indicate segment rejected
+			continue
+		}
+
+		// TODO: Transform payload to consumer.ConsumeTraceData. For now
```

Comment:
For this PR, maybe we could at least create pdata with name copied over or something? We'd like to avoid commented out code like `// payload := slices[1]`